code/game/machinery/landline.dm

```diff
@@ -0,0 +1,393 @@
+	
+/obj/landline/proc/has_power()
+	var/obj/machinery/requests_console/RC = attached_to
+	if(istype(RC, /obj/machinery/requests_console) && RC.stat)
+		return FALSE
+	return TRUE //redphones stay powered regardless
+	
+/obj/landline/proc/end_call_loop()
+	spawn(0)
+		while(linked_phone && !phone && has_power())
+			playsound(source=linked_phone, soundin=linked_phone.end_call_sound, vol=100, vary=FALSE, channel=0)
+			sleep(1 SECONDS)
+	
+/obj/landline/proc/ring_loop()
+	if(calling && calling.phone)
+		calling.ringing = TRUE
+	spawn(0)
+		while(calling && calling.calling == src && calling.ringing)
+			if(!has_power())
+				calling.ringing = FALSE
+			calling.ring()
+			if(linked_phone && !phone)
+				playsound(source=linked_phone, soundin=linked_phone.dial_sound, vol=100, vary=FALSE, channel=0)
+				//this sound is barely perceptible if you're not right next to the phone
+			sleep(5 SECONDS)
+
+/obj/landline/proc/start_call(var/obj/landline/destination)
+	if(!destination)
+		return "critical error"
+	if(calling)
+		return "you are already calling [calling.get_department()]"
+	if(destination.calling || !destination.phone)
+		end_call_loop()
+		return "line busy"
+	if(phone)
+		return "pick up the phone first"
+	for (var/obj/machinery/message_server/MS in message_servers)
+		if(MS.landlines_functioning())
+			calling = destination
+			destination.calling = src
+			ring_loop()
+			return "dialling..."
+	return "auto-routing offline. please wait for operator..."
+	
+			
+
+/obj/landline/proc/ring()
+	if(!linked_phone)
+		return
+	if(!attached_to)
+		return
+	if(!has_power())
+		return
+	if(ringer)
+		playsound(source=attached_to.loc, soundin=linked_phone.ringtone, vol=100, vary=FALSE, channel=CHANNEL_TELEPHONES)
+		shake_phone_overlay()
+
+/obj/landline/proc/pick_up_phone(mob/user)
+	if(!iscarbon(user))
+		to_chat(user, "<span class='notice'>You are not capable of such fine manipulation.</span>")
+		return
+	if(user.incapacitated())
+		to_chat(user, "<span class='notice'>You cannot do this while incapacitated.</span>")
+		return
+	if(!phone)
+		to_chat(user, "<span class='notice'>\the [src] has no telephone!</span>")
+		return
+	
+	user.put_in_hands(src.phone)
+	playsound(source=src, soundin= phone.pickup_sound, vol=100, vary=TRUE, channel=CHANNEL_TELEPHONES, wait=0)
+	phone = null //do not delete phone
+	attached_to.overlays.Remove(phone_overlay)
+	if(ringing && calling)
+		//TODO don't override the "rerouted by operator" line if we got rerouted
+		last_call_log = text("<B>Last call log:</B><BR><BR>")
+		last_call_log += text("picked up call from [calling.get_department()]<BR>")
+		calling.last_call_log += text("[get_department()] picked up call<BR>")
+		if(calling.linked_phone)
+			playsound(source=calling.linked_phone, soundin=linked_phone.pickup_sound, vol=100, vary=TRUE, channel=CHANNEL_TELEPHONES, wait=0)
+	ringing = FALSE
+	/* for(var/obj/machinery/computer/message_monitor/MM in message_monitors)

```

Comment:
Remove or complete this

Reply:
i tested this locally and it didn't work
but it may be because the window wasn't focused, since i had to switch users to test 2clients
gonna uncomment this unless there's something bad about this @Kurfursten

Reply:
There's no reason to expect this would be problematic, although obviously remove the message_admins testing statement